packages/cli/src/utils/init.js

```diff
@@ -28,6 +30,70 @@ const initApp = async (path, createFunc) => {
   await removeDir(tempAppDir);
 };
 
+const testAuth = authType => {
+  // todo: oauth1 trello url?
+  const testUrl =
+    authType === 'digest'
+      ? 'https://httpbin.org/digest-auth/auth/myuser/mypass'
+      : 'https://auth-json-server.zapier-staging.com/me';
+
+  return `
+      const test = (z, bundle) => {
+        return z.request({
```

Comment:
the tests don't do the check that they currently do - I've confirmed (both in local unit tests and connecting an auth on the site) that throwing for 401s works just as well in middleware as it does in the `testAuth` function. Not sure why an auth would only want to check for bad creds in the test function (and not always), but it seems fine here. 

If the test function throws, then that error will supersede any middleware.